src/main/java/uk/gov/hmcts/reform/pdf/service/aspect/AuthAspect.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.pdf.service.aspect;
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.pdf.service.service.AuthService;
+
+@Aspect
+@Component
+public class AuthAspect {
+
+    @Autowired
+    private AuthService authService;
+
+    @Before("execution(* uk.gov.hmcts.reform.pdf.service.endpoint.v2.*.*(String, ..))")
```

Comment:
I'm not sure about this approach. While keeping authentication in it's own place (which is nice), it's also quite a big assumption/restriction, which might be overlooked by developers adding new endpoints. What alternatives are there?

Reply:
It is exactly what's needed. Restriction is natural. Alternative is overviewed: another chain of interception